sdk/highlight-node/src/client.ts

```diff
@@ -212,4 +258,36 @@ export class Highlight {
 			.forceFlush()
 			.catch((e) => console.warn('highlight-node failed to flush: ', e))
 	}
+
+	async waitForFlush() {
+		return new Promise<void>(async (resolve) => {
+			let resolved = false
+			let waitingForFinishedSpans = false
+
+			await this.flush()
+
+			let intervalTimer = setInterval(async () => {
+				const finishedSpansCount = this.finishedSpans.length
+
+				if (finishedSpansCount) {
+					waitingForFinishedSpans = true
+				} else if (waitingForFinishedSpans) {
+					finish()
+				}
+			}, 100)
```

Comment:
any sense to make this smaller? i'd assume the spans to be inserted faster than that, since there isn't any network request to be done for that step

Reply:
We can do that. The `setInterval` spends most of its time waiting for the finished spans. Once it sees them, it can `finish` on the next tick.